modules/secrets-store-vault.adoc

```diff
@@ -0,0 +1,383 @@
+// Module included in the following assemblies:
+//
+// * nodes/pods/nodes-pods-secrets-store.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="secrets-store-vault_{context}"]
+= Mounting secrets from HashiCorp Vault
+
+You can use the {secrets-store-operator} to mount secrets from HashiCorp Vault to a CSI volume in {product-title}.
+
+// TODO: Is there a requirement for particular provider? https://issues.redhat.com/browse/OSDOCS-10136 steps say Azure, is this true?
+// To mount secrets from HashiCorp Vault, your cluster must be installed on Microsoft Azure.
+
+.Prerequisites
+
+// TODO
+// * Your cluster is installed on Azure.
+* You have installed the {secrets-store-operator}. See _Installing the {secrets-store-driver}_ for instructions.
+* You have installed Helm.
+* You have configured HashiCorp Vault to store the required secrets.
+* You have access to the cluster as a user with the `cluster-admin` role.
+
+.Procedure
+
+. Add the HashiCorp Helm repository by running the following command:
++
+[source,terminal]
+----
+$ helm repo add hashicorp https://helm.releases.hashicorp.com
+----
+
+. Update all repositories to ensure that Helm is aware of the latest versions by running the following command:
++
+[source,terminal]
+----
+$ helm repo update
+----
+
+. Create a new project for Vault by running the following command:
++
+[source,terminal]
+----
+$ oc new-project vault
+----
+
+. Install the HashiCorp Vault provider:
++
+////
+// TODO: Confirm if we still need this step to label the ns if we're doing a deployment instead of a pod
+.. Label the `vault` namespace to... TODO:
++
+[source,terminal]
+----
+$ oc label ns vault security.openshift.io/scc.podSecurityLabelSync=false pod-security.kubernetes.io/enforce=privileged pod-security.kubernetes.io/audit=privileged pod-security.kubernetes.io/warn=privileged --overwrite
+----
+////
+
+.. Grant privileged access to the `vault` service account by running the following command:
++
+[source,terminal]
+----
+$ oc adm policy add-scc-to-user privileged -z vault -n vault
+----
+
+.. Grant privileged access to the `vault-csi-provider` service account by running the following command:
++
+[source,terminal]
+----
+$ oc adm policy add-scc-to-user privileged -z vault-csi-provider -n vault
+----
+
+.. Deploy HashiCorp Vault by running the following command:
++
+[source,terminal]
+----
+$ helm install vault hashicorp/vault --namespace=vault \
```

Comment:
Can you plz update the steps:
   helm install vault hashicorp/vault --namespace=vault \
  --set "server.dev.enabled=true" \
  --set "injector.enabled=false" \
  --set "csi.enabled=true" \
  --set "global.openshift=true" \
  --set "injector.agentImage.repository=docker.io/hashicorp/vault" \
  --set "server.image.repository=docker.io/hashicorp/vault" \
  --set "csi.image.repository=docker.io/hashicorp/vault-csi-provider" \
  --set "csi.agent.image.repository=docker.io/hashicorp/vault" \
  --set "csi.daemonSet.providersDir=/var/run/secrets-store-csi-providers"